### ISSUE TITLE
phpcs (latest version in the repo)  goes into infinite loop (no response)

### ISSUE THREAD

**Issue by ftao** [squizlabs/PHP_CodeSniffer#563]
the file cause the problem  

https://raw.githubusercontent.com/Majkl578/doctrine2-slow-executors/master/Doctrine/ORM/Query/TreeWalker.php

The latest release 2.3.0  can process the file correctly .

**Comment by aik099** [comment]
What command you're using to run `phpcs`? Also what was standard name (if not specified in the command).

**Comment by ftao** [comment]
I'm not sure about the standard name ... 

I can reproduce the problem using following commands . 
```bash
wget https://raw.githubusercontent.com/Majkl578/doctrine2-slow-executors/master/Doctrine/ORM/Query/TreeWalker.php -O /tmp/TreeWalker.php
git clone https://github.com/squizlabs/PHP_CodeSniffer.git   /opt/PHP_CodeSniffer
/opt/PHP_CodeSniffer/scripts/phpcs  /tmp/TreeWalker.php
```

**Comment by gsherwood** [comment]
First, thank you very much for finding this before I released. I really appreciate you testing the master code.

I've committed a fix for it. I don't think it was actually an infinite loop, but it would have taken a huge amount of time to actually process that file. If you cut down the number of functions in the interface, it will complete, but it is basically going through the entire file each time a function is found. Obviously very bad.

Thanks again, and let me know if you have any other problems.
